pennylane/transforms/mitigate.py

```diff
@@ -538,6 +544,14 @@ def processing_fn(results):
 
         extrapolated = extrapolate(scale_factors, results_flattened, **extrapolate_kwargs)
 
+        if qml.active_return():
+            extrapolated = extrapolated[0] if shape(extrapolated) == (1,) else extrapolated
+
+            # unstack the results in the case of multiple measurements
+            return (
+                extrapolated if shape(extrapolated) == () else tuple(qml.math.unstack(extrapolated))
```

Comment:
Minor: can the two if statements checking `shape(extrapolated)` be merged, i.e., does it hold that the first in line 548 is `True` iff the second in line 552 is `True`? If that's the case, then worth simplifying the logic here. This is just a wild guess though.

Reply:
I also would have thought that, but it seems the shapes vary across the interfaces. For some the `extrapolated` variable has shape `()` at the start of this logic but some have shape `(1,)`.